backend/src/models/serviceTokenDataV3.ts

```diff
@@ -0,0 +1,129 @@
+import { Document, Schema, Types, model } from "mongoose";
+import { IPType } from "../ee/models";
+
+export enum Permission {
+    READ = "read",
+    WRITE = "write"
+}
+
+export interface IServiceTokenV3Scope {
+    environment: string;
+    secretPath: string;
+    permissions: Permission[];
+}
+
+export interface IServiceTokenV3TrustedIp {
+    ipAddress: string;
+    type: IPType;
+    prefix: number;
+}
+
+export interface IServiceTokenDataV3 extends Document {
+    _id: Types.ObjectId;
+    name: string;
+    workspace: Types.ObjectId;
+    user: Types.ObjectId;
+    publicKey: string;
+    isActive: boolean;
+    lastUsed?: Date;
+    usageCount: number;
+    expiresAt?: Date;
+    scopes: Array<IServiceTokenV3Scope>;
+    trustedIps: Array<IServiceTokenV3TrustedIp>;
+}
+
+const serviceTokenDataV3Schema = new Schema(
```

Comment:
I see that we have both a service token data v3 and service token data v3 key. Can you please explain high level how the crypto works for this?

Reply:
Yup different approach than V2.

Summary: ST V3 has a public-private key pair like a user. You share a copy of the project key with ST V3 by encrypting it under the public key of ST V3 - Exact same procedure as what we have for users and bots.

On the client-side, ST V3 decrypts the copy of the project key using its private key.